src/participant_network/cl/lighthouse/lighthouse_launcher.star

```diff
@@ -0,0 +1,300 @@
+
+#  ---------------------------------- Beacon client -------------------------------------
+CONSENSUS_DATA_DIRPATH_ON_BEACON_SERVICE_CONTAINER = "/consensus-data"
+
+# Port IDs
+BEACON_TCP_DISCOVERY_PORT_ID = "tcp-discovery"
+BEACON_UDP_DISCOVERY_PORT_ID = "udp-discovery"
+BEACON_HTTP_PORT_ID         = "http"
+BEACON_METRICS_PORT_ID      = "metrics"
+
+# Port nums
+BEACON_DISCOVERY_PORT_NUM = 9000
+BEACON_HTTP_PORT_NUM      = 4000
+BEACON_METRICS_PORT_NUM   = 5054
+
+#  ---------------------------------- Validator client -------------------------------------
+VALIDATING_REWARDS_ACCOUNT = "0x0000000000000000000000000000000000000000"
+
+VALIDATOR_HTTP_PORT_ID     = "http"
+VALIDATOR_METRICS_PORT_ID  = "metrics"
+VALIDATOR_HTTP_PORT_NUM    = 5042
+VALIDATOR_METRICS_PORT_NUM = 5064
+
+METRICS_PATH = "/metrics"
+
+BEACON_SUFFIX_SERVICE_ID    = "beacon"
+VALIDATOR_SUFFIX_SERVICE_ID = "validator"
+
+PRIVATE_IP_ADDRESS_PLACEHOLDER = "KURTOSIS_IP_ADDR_PLACEHOLDER"
+
+BEACON_USED_PORTS = {
+	BEACON_TCP_DISCOVERY_PORT_ID: new_port_spec(BEACON_DISCOVERY_PORT_NUM, TCP_PROTOCOL),
+	BEACON_UDP_DISCOVERY_PORT_ID: new_port_spec(BEACON_DISCOVERY_PORT_NUM, UDP_PROTOCOL),
+	BEACON_HTTP_PORT_ID:         new_port_spec(BEACON_HTTP_PORT_NUM, TCP_PROTOCOL),
+	BEACON_METRICS_PORT_ID:      new_port_spec(BEACON_METRICS_PORT_NUM, TCP_PROTOCOL),
+}
+
+VALIDATOR_USED_PORTS = {
+	VALIDATOR_HTTP_PORT_ID:    new_port_spec(VALIDATOR_HTTP_PORT_NUM, TCP_PROTOCOL),
+	VALIDATOR_METRICS_PORT_ID: new_port_spec(VALIDATOR_METRICS_PORT_NUM, TCP_PROTOCOL),
+}
+
+LIGHTHOUSE_LOG_LEVELS = {
+	module_io.GlobalClientLogLevel.error: "error",
+	module_io.GlobalClientLogLevel.warn:  "warn",
+	module_io.GlobalClientLogLevel.info:  "info",
+	module_io.GlobalClientLogLevel.debug: "debug",
+	module_io.GlobalClientLogLevel.trace: "trace",
+}
+
+BEACON_ENR_FACT_NAME = "beacon-enr-fact"
+BEACON_HEALTH_FACT_NAME = "beacon-health-fact"
+
+def launch(
+	launcher,
+	service_id,
+	image,
+	participant_log_level,
+	global_log_level,
+	bootnode_context,
+	el_client_context,
+	mev_boost_context,
+	node_keystore_files,
+	extra_beacon_params,
+	extra_validator_params):
+
+	beacon_node_service_id = "{0}-{1}".format(service_id, BEACON_SUFFIX_SERVICE_ID)
+	validator_node_service_id = "{0}-{1}".format(service_id, VALIDATOR_SUFFIX_SERVICE_ID)
+
+	log_level = get_client_log_level_or_default(participant_log_level, global_log_level, LIGHTHOUSE_LOG_LEVELS)
+
+	# Launch Beacon node
+	beacon_config = get_beacon_config(
+		launcher.genesis_data,
+		image,
+		bootnode_context,
+		el_client_context,
+		mev_boost_context,
+		log_level,
+		extra_beacon_params,
+	)
+
+	beacon_service = add_service(beacon_node_service_id, beacon_config)
+
+	# TODO check whether its 200, 206 or 503 like golang
+	define_fact(service_id = beacon_node_service_id, fact_name = BEACON_HEALTH_FACT_NAME, fact_recipe = struct(method= "GET", endpoint = "/eth/v1/node/health", content_type = "application/json", port_id = BEACON_HTTP_PORT_ID))
```

Comment:
We should maybe create our own named struct for those object. `fact_recipe(method = "GET", ...)` would look a bit more ironed IMO

Reply:
@victorcolombo  is reworking how this works underneath 
https://www.notion.so/kurtosistech/One-off-HTTP-Request-cf290802339b4753b4bdde0946e1cd31